src/main/java/com/metamx/common/StreamUtils.java

```diff
@@ -35,6 +37,7 @@
   // The default buffer size to use (from IOUtils)
   private static final int DEFAULT_BUFFER_SIZE = 1024 * 4;
 
+  @Deprecated // Use FileUtils.retryCopy
```

Comment:
The goal was to try and stop using InputStream and instead use ByteSource so that retries can even be attempted. Using InputStream in a way such that the operation cannot be attempted again is something IMHO we should try to stop in general. That is why I deprecated this.